src/Microsoft.ML.Data/DataLoadSave/TransformWrapper.cs

```diff
@@ -23,99 +16,34 @@ namespace Microsoft.ML.Data
     internal sealed class TransformWrapper : ITransformer
     {
         internal const string LoaderSignature = "TransformWrapper";
-        private const string TransformDirTemplate = "Step_{0:000}";
 
         private readonly IHost _host;
         private readonly IDataView _xf;
-        private readonly bool _allowSave;
-        private readonly bool _isRowToRowMapper;
 
-        public TransformWrapper(IHostEnvironment env, IDataView xf, bool allowSave = false)
+        public TransformWrapper(IHostEnvironment env, IDataView xf)
         {
             Contracts.CheckValue(env, nameof(env));
+            Contracts.Check(xf is IDataTransform);
+
             _host = env.Register(nameof(TransformWrapper));
             _host.CheckValue(xf, nameof(xf));
+            _host.Check(IsChainRowToRowMapper(_xf));
```

Comment:
>IsChainRowToRowMapper [](start = 24, length = 21)

Do we need to check this?
Or is it enough to just check that _xf is `IRowToRowMapper`?
Or maybe even to not check it, and return `_xf is IRowToRowMapper` instead of `true` for the `IsRowToRowMapper` property?